### PR TITLE
Add check for [wip] in PR title

### PR DESCRIPTION
**Pull request by Shane32**
See #1901 

Note: checks should fail until removing the text "WIP" in the title of this PR.

@sungam3r Try it and see.

### CODE REVIEW

File: .github/workflows/wipcheck.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      - edited        # when PR title is edited
+      - synchronize   # when code is added
+      - reopened      # when a closed PR is reopened
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail if pull request title '${{ github.event.pull_request.title }}' contains 'WIP'
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:16]:
```suggestion
      - name: Fail build if pull request title contains 'WIP'
```

**Comment by sungam3r** [comment]
изображение

Could you please suppress duplicated  echo?

File: .github/workflows/wipcheck.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        if: ${{ contains(github.event.pull_request.title, 'wip') }}
+        run: |
+          echo Error! PR title contains WIP: ${{ github.event.pull_request.title }}
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:19]:
```suggestion
          echo Warning! PR title '${{ github.event.pull_request.title }}' contains WIP. Remove WIP from the title when PR is ready.
```

File: .github/workflows/wipcheck.yml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+name: Check if PR title contains WIP
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:1]:
```suggestion
name: Check if PR title contains [WIP]
```

File: .github/workflows/wipcheck.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      - edited        # when PR title is edited
+      - synchronize   # when code is added
+      - reopened      # when a closed PR is reopened
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail build if pull request title contains 'WIP'
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:16]:
```suggestion
      - name: Fail build if pull request title contains '[WIP]'
```

File: .github/workflows/wipcheck.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+        if: ${{ contains(github.event.pull_request.title, 'wip') }}
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:17]:
```suggestion
        if: ${{ contains(github.event.pull_request.title, '[wip]') }} # This function is not case sensitive.
```

File: .github/workflows/wipcheck.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+        run: |
+          echo Warning! PR title '${{ github.event.pull_request.title }}' contains WIP. Remove WIP from the title when PR is ready.
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:19]:
```suggestion
          echo Warning! PR title '${{ github.event.pull_request.title }}' contains [WIP]. Remove [WIP] from the title when PR is ready.
```

File: .github/workflows/wipcheck.yml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      - edited        # when PR title is edited
+      - synchronize   # when code is added
+      - reopened      # when a closed PR is reopened
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail build if pull request title contains '[WIP]'
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:16]:
```suggestion
      - name: Fail build if pull request title contains [WIP]
```

File: .github/workflows/wipcheck.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+        if: ${{ contains(github.event.pull_request.title, '[wip]') }} # This function is not case sensitive.
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:17]:
```suggestion
        if: ${{ contains(github.event.pull_request.title, '[WIP]') }} # This function is not case sensitive.
```

File: .github/workflows/wipcheck.yml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail build if pull request title contains [WIP]
+        if: ${{ contains(github.event.pull_request.title, '[WIP]') }} # This function is not case sensitive.
+        run: |
+          echo Warning! PR title '${{ github.event.pull_request.title }}' contains [WIP]. Remove [WIP] from the title when PR is ready.
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:19]:
```suggestion
          echo Warning! PR title ''${{ github.event.pull_request.title }}'' contains [WIP]. Remove [WIP] from the title when PR is ready.
```

File: .github/workflows/wipcheck.yml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+jobs:
+  check-title:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail build if pull request title contains [WIP]
+        if: ${{ contains(github.event.pull_request.title, '[WIP]') }} # This function is not case sensitive.
+        run: |
+          echo Warning! PR title ''${{ github.event.pull_request.title }}'' contains [WIP]. Remove [WIP] from the title when PR is ready.
```
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:19]:
```suggestion
          echo Warning! PR title '${{ github.event.pull_request.title }}' contains [WIP]. Remove [WIP] from the title when PR is ready.
```

**Comment by sungam3r** [comment]
изображение
😞 

Well, OK. Thanks.

**Comment by Shane32** [comment]
Maybe double quotes would work.  Not that it really matters.

**Comment by Shane32** [comment]
I don't know enough about the ubuntu shell (bash?) to know what's going on.  I never use linux.

File: .github/workflows/wipcheck.yml (hunk 9 above)
Comment (sungam3r) [inline .github/workflows/wipcheck.yml:19]:
```suggestion
          echo Warning! PR title "${{ github.event.pull_request.title }}" contains [WIP]. Remove [WIP] from the title when PR is ready.
```

**Comment by sungam3r** [comment]
Me too. Just interesting about quotes.

**Comment by Shane32** [comment]
https://unix.stackexchange.com/questions/30903/how-to-escape-quotes-in-shell

**Comment by sungam3r** [comment]
изображение

😠

**Comment by sungam3r** [comment]
I tried to change the description of two PRs with [WIP] in title, but the check did not work.

**Comment by Shane32** [comment]
Probably because they are old PRs and the branch does not contain the workflow.

**Pull request merged**